File: lib/models/animal_case.dart
enum CaseStatus {
  reported,
  facilitySelected,
  admitted,
  underTreatment,
  fundingReview,
  treatmentCompleted,
  shelterCare,
  readyForAdoption,
  adopted,
}

extension CaseStatusExtension on CaseStatus {
  String get displayName {
    switch (this) {
      case CaseStatus.reported:
        return 'Reported';
      case CaseStatus.facilitySelected:
        return 'Facility Selected';
      case CaseStatus.admitted:
        return 'Admitted';
      case CaseStatus.underTreatment:
        return 'Under Treatment';
      case CaseStatus.fundingReview:
        return 'Funding Review';
      case CaseStatus.treatmentCompleted:
        return 'Treatment Completed';
      case CaseStatus.shelterCare:
        return 'Shelter Care';
      case CaseStatus.readyForAdoption:
        return 'Ready for Adoption';
      case CaseStatus.adopted:
        return 'Adopted';
    }
  }

  int get stepIndex {
    switch (this) {
      case CaseStatus.reported:
        return 0;
      case CaseStatus.facilitySelected:
        return 1;
      case CaseStatus.admitted:
        return 2;
      case CaseStatus.underTreatment:
        return 3;
      case CaseStatus.fundingReview:
        return 4;
      case CaseStatus.treatmentCompleted:
        return 5;
      case CaseStatus.shelterCare:
        return 6;
      case CaseStatus.readyForAdoption:
        return 6;
      case CaseStatus.adopted:
        return 7;
    }
  }
}

class AnimalCase {
  final String id;
  final String animalType; // Dog, Cat, Other
  final String condition; // Critical, Injured, Sick, Healthy, Abandoned, Puppy/Kitten
  final String description;
  final String ageGroup; // Young, Adult, Senior, Unknown
  final String locationAddress;
  final String landmark;
  final String volunteerAction; // Transport, Need Help, Temp Care
  final String selectedFacilityId;
  final String selectedFacilityName;
  CaseStatus status;
  final String reporterName;
  final String reporterPhone;
  final String imageUrl;
  final DateTime createdAt;
  final double estimatedCost;
  final double pawcareSupport;
  final double donorSupport;
  final double remainingCost;
  final List<String> medicalNotes;

  AnimalCase({
    required this.id,
    required this.animalType,
    required this.condition,
    required this.description,
    required this.ageGroup,
    required this.locationAddress,
    required this.landmark,
    required this.volunteerAction,
    required this.selectedFacilityId,
    required this.selectedFacilityName,
    required this.status,
    required this.reporterName,
    required this.reporterPhone,
    required this.imageUrl,
    required this.createdAt,
    this.estimatedCost = 0.0,
    this.pawcareSupport = 0.0,
    this.donorSupport = 0.0,
    this.remainingCost = 0.0,
    this.medicalNotes = const [],
  });
}

File: animal-connect-backend/src/models/AnimalCase.js
const mongoose = require('mongoose');

const animalCaseSchema = new mongoose.Schema(
  {
    caseId: {
      type: String,
      required: true,
      unique: true,
      index: true
    },
    reporter: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User',
      required: true
    },
    reporterName: {
      type: String,
      required: true
    },
    reporterPhone: {
      type: String,
      required: true
    },
    animalType: {
      type: String,
      required: [true, 'Animal type is required (e.g. Dog, Cat, Other)'],
      trim: true
    },
    condition: {
      type: String,
      enum: ['Critical', 'Injured', 'Sick', 'Healthy', 'Abandoned', 'Puppy/Kitten'],
      required: true
    },
    description: {
      type: String,
      required: [true, 'Case description is required'],
      trim: true
    },
    ageGroup: {
      type: String,
      enum: ['Young', 'Adult', 'Senior', 'Unknown'],
      default: 'Unknown'
    },
    gender: {
      type: String,
      enum: ['Male', 'Female', 'Unknown'],
      default: 'Unknown'
    },
    locationAddress: {
      type: String,
      required: [true, 'Location address is required'],
      trim: true
    },
    landmark: {
      type: String,
      trim: true,
      default: ''
    },
    location: {
      type: {
        type: String,
        enum: ['Point'],
        default: 'Point'
      },
      coordinates: {
        type: [Number], // [longitude, latitude]
        default: [76.9558, 11.0168]
      }
    },
    images: [
      {
        url: { type: String, required: true },
        publicId: { type: String },
        caption: { type: String, default: '' },
        uploadedBy: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
        uploadedRole: { type: String, default: 'volunteer' },
        createdAt: { type: Date, default: Date.now }
      }
    ],
    volunteerAction: {
      type: String,
      default: 'Need Help'
    },
    selectedFacility: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'Facility'
    },
    selectedFacilityName: {
      type: String,
      default: ''
    },
    status: {
      type: String,
      enum: [
        'Reported',
        'Facility Selected',
        'Admitted',
        'Under Treatment',
        'Funding Review',
        'Treatment Completed',
        'Shelter Care',
        'Ready for Adoption',
        'Adopted',
        'Closed'
      ],
      default: 'Reported',
      index: true
    },
    priority: {
      type: String,
      enum: ['Low', 'Medium', 'High', 'Critical'],
      default: 'Medium'
    },
    estimatedCost: {
      type: Number,
      default: 0
    },
    pawcareSupport: {
      type: Number,
      default: 0
    },
    donorSupport: {
      type: Number,
      default: 0
    },
    remainingCost: {
      type: Number,
      default: 0
    },
    medicalNotes: [
      {
        type: String
      }
    ],
    photoMatchStatus: {
      type: String,
      enum: ['pending', 'matched', 'mismatch', 'manual_review'],
      default: 'pending'
    },
    photoMatchNotes: {
      type: String,
      default: ''
    }
  },
  {
    timestamps: true,
    toJSON: { virtuals: true },
    toObject: { virtuals: true }
  }
);

// Indexes
animalCaseSchema.index({ location: '2dsphere' });
animalCaseSchema.index({ status: 1, condition: 1, priority: 1 });

// Helper virtual for image URL to match Flutter single imageUrl field
animalCaseSchema.virtual('imageUrl').get(function () {
  if (this.images && this.images.length > 0) {
    return this.images[0].url;
  }
  return 'https://images.unsplash.com/photo-1543466835-00a7907e9de1?w=600';
});

const AnimalCase = mongoose.model('AnimalCase', animalCaseSchema);
module.exports = AnimalCase;
